gemrb/plugins/SDLVideo/Shaders/BlitRGB.glsl

```diff
@@ -0,0 +1,31 @@
+precision highp float;
+
+varying vec2 v_texCoord;
+varying vec4 v_color;
+
+uniform sampler2D s_sprite;
+uniform sampler2D s_stencil;
+uniform int u_greyMode;
+uniform int u_channel;
+uniform int u_stencil;
+
+void main() {
+	vec4 color = texture2D(s_sprite, v_texCoord);
+	gl_FragColor = color;
+	gl_FragColor.a = 1.0;
+	gl_FragColor *= v_color;
```

Comment:
shouldn't we be multiplying by `v_color` in all cases? the way its written now this will be undone for `u_greyMode`s

I also wonder if there is a noticeable performance difference if we had an `else` instead of setting `gl_FragColor` multiple times.

Reply:
> shouldn't we be multiplying by `v_color` in all cases? the way its written now this will be undone for `u_greyMode`s

I copied that from the original `SHADER_RGB` code https://github.com/libsdl-org/SDL/blob/1868c5b5210636530d6945c0ec16b11e3506248d/src/render/opengl/SDL_shaders_gl.c#L266-L270 and suspect that we need to keep it unless we want do dome some RE to also maintain color channels aligned with SDL's draw state. Otherwise, I'll probably be failing with something like the black areas as initially.

Partially, yes. But I've hardcoded the alpha to `1.0` there, too. But see below ...

> I also wonder if there is a noticeable performance difference if we had an `else` instead of setting `gl_FragColor` multiple times.

Lecture always tells you -- I cannot verify this -- that if-branching in shaders does not lead to selected execution but always ends up rolling out every code line into the shader cores with discard masks. Anyway, I guess this tiny little shader is even too trivial for small platforms, and all the profiling tools I'm aware of only work with OpenGL v3.1+